cmd/goprotoc/options.go

```diff
@@ -139,8 +139,11 @@ func parseFlags(source string, args []string, opts *protocOptions, sourcesSeen m
 						return err
 					}
 				}
-			case strings.HasPrefix(a, "--") && strings.HasSuffix(a, "_out"):
-				opts.output[a[2:len(a)-4]] = getOptionArg()
+			case strings.HasPrefix(parts[0], "--") && strings.HasSuffix(parts[0], "_out"):
+				if opts.output == nil {
```

Comment:
Doh! You can tell how little testing I did (it's almost embarrassing!). I mostly played with the esoteric features -- like finding free field numbers and doing the encode/decode actions. I think I also tested saving to a descriptor set.

Thanks for putting it through the ringer!

Reply:
No problem! I'd love to set up a large-scale integration test for this that takes a bunch of well-known repos (googleapis, etcd perhaps, etc) and does a version of my verification in an automated fashion, so that `goprotoc` is always up to date. This would be easier if `goprotoc` was part of `protoreflect` directly, since it's changes to `protoreflect` that will have the most effect.

FYI, the script I've been using to get to this, running this from the root of googleapis:

```
#!/usr/bin/env bash

set -exo pipefail

DIR="$(cd "$(dirname "${0}")" && pwd)"
cd "${DIR}"

LANGS="java python"

prototool compile --protoc-bin-path="$(which goprotoc)" --protoc-wkt-path=/usr/local/include
prototool compile --protoc-bin-path="$(which protoc)" --protoc-wkt-path=/usr/local/include

rm -rf gen
mkdir -p gen
time prototool descriptor-set --include-imports --protoc-bin-path="$(which goprotoc)" --protoc-wkt-path=/usr/local/include --json | jq . > gen/goprotoc_descriptor_set.json
time prototool descriptor-set --include-imports --protoc-bin-path="$(which protoc)" --protoc-wkt-path=/usr/local/include --json | jq . > gen/protoc_descriptor_set.json
diff gen/goprotoc_descriptor_set.json gen/protoc_descriptor_set.json

mkdir -p gen/protoc gen/goprotoc
time prototool generate --protoc-bin-path="$(which goprotoc)" --protoc-wkt-path=/usr/local/include
for lang in ${LANGS}; do
  mv gen/${lang} gen/goprotoc/${lang}
done
time prototool generate --protoc-bin-path="$(which protoc)" --protoc-wkt-path=/usr/local/include
for lang in ${LANGS}; do
  mv gen/${lang} gen/protoc/${lang}
done
for lang in ${LANGS}; do
  cd gen/goprotoc && find ${lang} -type f | sort | uniq > ../goprotoc_${lang}_files.txt && cd -
  cd gen/protoc && find ${lang} -type f | sort | uniq > ../protoc_${lang}_files.txt && cd -
  diff gen/goprotoc_${lang}_files.txt gen/protoc_${lang}_files.txt
done

diff_files() {
  set +x
  mkdir -p "gen/diff${1}"
  while IFS= read -r line; do
    echo "$line"
    mkdir -p "$(dirname "gen/diff${1}/$line")"
    set +e
    diff "gen/goprotoc/$line" "gen/protoc/$line" > "gen/diff${1}/$line.diff"
    set -e
  done < "gen/goprotoc_$1_files.txt"
  set -x
}

diff_files java
diff_files python
```

You need `prototool` installed and to have this `prototool.yaml` at the root:

```
protoc:
  allow_unused_imports: true
```

Reply:
I'm a little hesitant to collapse this repo with the protoreflect repo since I feel like this falls too far from "reflection".

If using Go modules, then I think it makes sense to have the tests here, so they can run whenever updating goprotoc's `go.mod` to point to a different/newer version of protoreflect. (Speaking of which, I'll need to update this repo to use new protoreflect release. I guess after this PR, I can also go ahead and setup goreleaser config for this repo and create a v0.1 of goprotoc, too!)